commandapi-platforms/commandapi-bukkit/commandapi-bukkit-test/commandapi-bukkit-test-tests/src/test/java/dev/jorel/commandapi/test/CommandNamespaceTests.java

```diff
@@ -0,0 +1,767 @@
+				.then(new IntegerArgument("integer")
+					.executesPlayer(info -> {
+						results.set("b");
+						results.set(String.valueOf(info.args().get("integer")));
+					})
+				)
+			);
+
+		// Make sure the default registration with the minecraft: namespace works
+		command.register();
+
+		if (!enableBeforeRegistering) {
+			player = enableWithNamespaces();
+		}
+
+		server.dispatchCommand(player, "test a alpha");
+		assertEquals("a", results.get());
+		assertEquals("alpha", results.get());
+
+		server.dispatchCommand(player, "minecraft:test a alpha");
+		assertEquals("a", results.get());
+		assertEquals("alpha", results.get());
+
+		server.dispatchCommand(player, "test b 123");
+		assertEquals("b", results.get());
+		assertEquals("123", results.get());
+
+		server.dispatchCommand(player, "minecraft:test b 123");
+		assertEquals("b", results.get());
+		assertEquals("123", results.get());
+
+		CommandAPI.unregister("test", true);
+
+		command.register("namespace");
+
+		server.dispatchCommand(player, "test a alpha");
+		assertEquals("a", results.get());
+		assertEquals("alpha", results.get());
+
+		server.dispatchCommand(player, "namespace:test a alpha");
+		assertEquals("a", results.get());
+		assertEquals("alpha", results.get());
+
+		server.dispatchCommand(player, "test b 123");
+		assertEquals("b", results.get());
+		assertEquals("123", results.get());
+
+		server.dispatchCommand(player, "namespace:test b 123");
+		assertEquals("b", results.get());
+		assertEquals("123", results.get());
+
+		// Running the command with the minecraft: namespace should fail
+		assertCommandFailsWith(
+			player,
+			"minecraft:test a alpha",
+			"Unknown or incomplete command, see below for error at position 0: <--[HERE]"
+		);
+		assertCommandFailsWith(
+			player,
+			"minecraft:test b 123",
+			"Unknown or incomplete command, see below for error at position 0: <--[HERE]"
+		);
+
+		CommandAPI.unregister("test", true);
+
+		command.register(MockPlatform.getConfiguration().getPlugin());
+
+		server.dispatchCommand(player, "test a alpha");
+		assertEquals("a", results.get());
+		assertEquals("alpha", results.get());
+
+		server.dispatchCommand(player, "commandapitest:test a alpha");
+		assertEquals("a", results.get());
+		assertEquals("alpha", results.get());
+
+		server.dispatchCommand(player, "test b 123");
+		assertEquals("b", results.get());
+		assertEquals("123", results.get());
+
+		server.dispatchCommand(player, "commandapitest:test b 123");
+		assertEquals("b", results.get());
+		assertEquals("123", results.get());
+
+		// Running the command with the minecraft: namespace should fail
+		assertCommandFailsWith(
+			player,
+			"minecraft:test a alpha",
+			"Unknown or incomplete command, see below for error at position 0: <--[HERE]"
+		);
+		assertCommandFailsWith(
+			player,
+			"minecraft:test b 123",
+			"Unknown or incomplete command, see below for error at position 0: <--[HERE]"
+		);
+
+		assertNoMoreResults(results);
+	}
+
+	@Disabled
```

Comment:
Please, go ahead. They do work in-game, I did verify that so that's nothing that should stop us from merging.
Figuring this out would be great though!